duffy/api_v2/views.py

```diff
@@ -0,0 +1,206 @@
+# -*- coding: utf-8 -*-
+import datetime
+
+from flask import Blueprint, request, jsonify, abort, current_app
+from functools import wraps
+from duffy.models import Openstack_Host, HostSchema, Session, SessionSchema, Project
+from duffy.database import db
+from credentials import get_nova_credentials
+from novaclient.Client import client
+
+blueprint = Blueprint('api_v1', __name__)
+
+
+def duffy_key_required(fn):
+    @wraps(fn)
+    def decorated(*args, **kwargs):
+        duffy_key = request.args.get('key', None)
+        if not duffy_key:
+            return jsonify({'msg': 'Invalid duffy key'}), 403
+        return fn(*args, **kwargs)
+    return decorated
+
+def ssid_required(fn):
+    @wraps(fn)
+    def decorated(*args, **kwargs):
+        ssid = request.args.get('ssid')
+        if not ssid:
+            return jsonify({'msg': 'Invalid session ID'}), 403
+
+        sess = Session.query.get(ssid)
+        if not sess:
+            return jsonify({'msg': 'Invalid session ID'}), 403
+        return fn(*args, **kwargs)
+    return decorated
+
+
+@blueprint.route('/Node/get')
+@duffy_key_required
+def nodeget():
```

Comment:
Let's be sure to handle the `type` argument (default is 'vm', if something else is provided then fail for now). This method is probably going to need to be refactored once we get around to adding bare metal support in these api calls.